features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/ConsentsHostFragment.kt

```diff
@@ -2,24 +2,33 @@ package `in`.projecteka.jataayu.consent.ui.fragment
 
 import `in`.projecteka.jataayu.consent.databinding.FragmentConsentHostBinding
 import `in`.projecteka.jataayu.consent.ui.adapter.ConsentPagerAdapter
+import `in`.projecteka.jataayu.consent.viewmodel.ConsentHostFragmentViewModel
 import `in`.projecteka.jataayu.core.model.MessageEventType
 import `in`.projecteka.jataayu.presentation.ui.fragment.BaseFragment
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.lifecycle.Observer
 import androidx.viewpager.widget.ViewPager
+import com.google.android.material.tabs.TabLayout
 import kotlinx.android.synthetic.main.activity_consent.*
 import org.greenrobot.eventbus.EventBus
 import org.greenrobot.eventbus.Subscribe
 import org.greenrobot.eventbus.ThreadMode
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
 
 class ConsentHostFragment : BaseFragment() {
 
     private lateinit var binding: FragmentConsentHostBinding
 
     private val eventBusInstance = EventBus.getDefault()
 
+    private val viewModel: ConsentHostFragmentViewModel by sharedViewModel()
+
+    private lateinit var viewPager: ViewPager
```

Comment:
we don't require to have a reference since we can access it from the binding. This way, we ensure we have ids for the elements.